lib/src/models/feedItem.dart

```diff
@@ -0,0 +1,65 @@
+import 'homeworkItem.dart';
+import 'meeting.dart';
+import 'sectionItem.dart';
+
+enum FeedItemType { Homework, Lecture, Section }
+
+extension FeedItemTypeExtension on FeedItemType {
+  static FeedItemType? fromItem(dynamic item) {
+    if (item is HomeworkItem) {
+      return FeedItemType.Homework;
+    } else if (item is SectionItem) {
+      return FeedItemType.Section;
+    }
+  }
+
+  static FeedItemType? fromIndex(int index) {
+    return FeedItemType.values[index];
+  }
+}
+
+class FeedItem {
+  /// The owning section for this feed item
+  SectionItem owner;
+
+  /// The "body" of this feed item. Description text/message
+  String body;
+
+  /// The post date for this feed item
+  DateTime postDate;
+
+  /// The item type for this feed item (to differentiate how it displays/links to its item)
+  FeedItemType type;
+
+  /// The action undertaken to create this feed item
+  String action;
+
+  /// The code of the item referenced by this feed item
+  String itemCode;
```

Comment:
@jwparktom @AndrewLester Perhaps we could have the API return a corresponding `SectionItem` instead of an `itemCode`, that way we don't need to create another request when opening the section from this `FeedItem`?

Reply:
I can cover this a bit more tomorrow, but using the SectionItem returned in the feed list requests (as it currently returns sections as well) was unfortunately not enough, since the section view also requires a course.

We could return sections AND courses from the feed endpoint. It may be better to discuss this further tomorrow.

Reply:
Oh yeah, I forgot about that.... hmm... If I'm not mistaken, the only use for the `CourseItem` was to show the title, perhaps we can get that from a `SectionItem` itself by changing the model?

Reply:
That's a fair idea, my only issue with that is we may not need it in other cases, so it would appear to be an odd piece of data to leave in the SectionItem. It may already be in the section item though to be honest. In fact, I think it *is*, inside the section's `full_code`. We can talk about this later, maybe we should write some helper methods to turn full codes -> subject ID/course ID